File: ObjC/SVSAVPlayerSampleWithBumper/SVSAVPlayerSampleWithBumper/VideoContainerView.m
//
//  VideoContainerView.m
//  SVSAVPlayerSample
//
//  Created by glaubier on 17/01/2018.
//  Copyright © 2018 Smart AdServer. All rights reserved.
//

#import "VideoContainerView.h"

@implementation VideoContainerView

+ (Class)layerClass {
    return [AVPlayerLayer class];
}

- (AVPlayer *)player {
    return [(AVPlayerLayer *)[self layer] player];
}

- (void)setPlayer:(AVPlayer *)player {
    [(AVPlayerLayer *)[self layer] setPlayer:player];
}

@end

File: Swift/SVSAVPlayerSampleWithBumper/SVSAVPlayerSampleWithBumper/Player/VideoContainerView.swift
//
//  VideoContainerView.swift
//  SVSAVPlayerSample
//
//  Created by Loïc GIRON DIT METAZ on 28/08/2019.
//  Copyright © 2019 Equativ. All rights reserved.
//

import UIKit
import AVFoundation

class VideoContainerView: UIView {
    
    override class var layerClass: AnyClass {
        return AVPlayerLayer.self
    }
    
    var player: AVPlayer {
        get {
            return (layer as! AVPlayerLayer).player!
        }
        set {
            (layer as! AVPlayerLayer).player = newValue
        }
    }
    
}
